### ISSUE TITLE
Scanner not found

### ISSUE THREAD

**Issue by 3like3beer** [openpaperwork/paperwork#678]
May be duplicate of issue #632 
On Ubuntu16.04 lts (Linux PC 4.4.0-93-generic #116-Ubuntu SMP Fri Aug 11 21:17:52 UTC 2017 i686 i686 i686 GNU/Linux)

Scanner not found error.

I can access the scanner with simple scan.
**Logs :**
INFO   paperwork.frontend.util.actions Action: [Scan a single page]
INFO   paperwork.frontend.util.config Will scan using hpaio:/net/HP_LaserJet_Pro_MFP_M127fn?ip=192.168.1.14
INFO   paperwork.frontend.util.config Will scan at a resolution of 300
INFO   paperwork.frontend.util.config Will scan using source ADF
INFO   paperwork.frontend.util.config Scanner mode set to 'Color'
WARNING paperwork.frontend.mainwindow  Exception while configuring scanner: <type 'exceptions.StopIteration'>: . Assuming scanner is not connected
INFO   paperwork.frontend.util.dialog Showing popup !
INFO   paperwork.frontend.util.actions Action: [Quit (real)]
INFO   paperwork.frontend.util.actions Action: [Open selected document]
INFO   paperwork.frontend.mainwindow  No document selected. Can't open

**more /home/julien/.config/paperwork.conf**
[GUI]
sorting = scan_date
main_win_size_w = 1375
main_win_size_h = 876
zoom_level = 0.25

[Global]
workdirectory = /home/julien/papers

[OCR]
lang = fra
enabled = True

[Scanner]
device = hpaio:/net/HP_LaserJet_Pro_MFP_M127fn?ip=192.168.1.14
source = ADF
resolution = 300
scantimecalibration = 11.4988150597
calibration_resolution = 200
calibration_pt_a_x = 0
calibration_pt_a_y = 0
calibration_pt_b_x = 1699
calibration_pt_b_y = 2337

[SpellChecking]
lang = None

**Comment by jflesch** [comment]
Not a duplicate. Scanner support if entirely different between Windows and Linux. #632 is about Windows.

**Comment by jflesch** [comment]
By the way, I see you selected ADF (feeder) as default source. Did you try scanning with the Flatbed ? Does it work ?

**Comment by 3like3beer** [comment]
Id did try with flatbed same issue (just did it  again, same problem)

**Comment by jflesch** [comment]
Can I have the logs with the Flatbed please ? Hopefully the output is slightly different and will give me a hint about what's going on.

**Comment by 3like3beer** [comment]
Sure :
When I press Scan :
INFO   paperwork.frontend.util.actions Action: [Scan a single page]
INFO   paperwork.frontend.util.config Will scan using hpaio:/net/HP_LaserJet_Pro_MFP_M127fn?ip=192.168.1.14
INFO   paperwork.frontend.util.config Will scan at a resolution of 300
WARNING paperwork.frontend.mainwindow  Exception while configuring scanner: <class 'pyinsane.rawapi.SaneException'>: <class 'pyinsane.rawapi.SaneStatus'> : Error during device I/O (9). Assuming scanner is not connected
INFO   paperwork.frontend.util.dialog Showing popup !
INFO   paperwork.frontend.util.actions Action: [Open settings dialog]
INFO   root                           UI file used: /usr/share/paperwork/settingswindow/settingswindow.glade
INFO   paperwork.frontend.settingswindow Distribution: [ubuntu]
ERROR  paperwork.frontend.settingswindow Warning: Long name not found for language 'osd'.
WARNING paperwork.frontend.settingswindow   Will use short name as long name.
ERROR  paperwork.frontend.settingswindow Warning: Long name not found for language 'equ'.
WARNING paperwork.frontend.settingswindow   Will use short name as long name.
INFO   paperwork.frontend.util.jobs   [Scheduler progress] Starting


When I check the printer 

INFO   paperwork.frontend.util.jobs   [Scheduler progress] Started
INFO   paperwork.frontend.settingswindow Looking for scan devices ...
INFO   paperwork.frontend.settingswindow Device found: [Hewlett-Packard HP_LaserJet_Pro_MFP_M127fn] -> [hpaio:/net/HP_LaserJet_Pro_MFP_M127fn?ip=192.168.1.200]
INFO   paperwork.frontend.settingswindow Got value [['Hewlett-Packard HP_LaserJet_Pro_MFP_M127fn', 'hpaio:/net/HP_LaserJet_Pro_MFP_M127fn?ip=192.168.1.200']]
WARNING paperwork.frontend.util.jobs   Job DeviceFinder:0 took 6086ms and is unstoppable ! (maximum allowed: 500ms)
INFO   paperwork.frontend.settingswindow Selected scanner: 0
INFO   paperwork.frontend.settingswindow Looking for resolution of device [hpaio:/net/HP_LaserJet_Pro_MFP_M127fn?ip=192.168.1.200]
/usr/local/lib/python2.7/dist-packages/paperwork/paperwork.py:111: Warning: unable to set property 'xpad' of type 'guint' from value of type 'gchararray'
  Gtk.main()
INFO   paperwork.frontend.settingswindow Sources found: ['Flatbed', 'ADF']
INFO   paperwork.frontend.settingswindow Got value [['Plateau', 'Flatbed']]
WARNING paperwork.frontend.util.jobs   Job SourceFinder:0 took 3772ms and is unstoppable ! (maximum allowed: 500ms)
INFO   paperwork.frontend.settingswindow Looking for resolution of device [hpaio:/net/HP_LaserJet_Pro_MFP_M127fn?ip=192.168.1.200]
INFO   paperwork.frontend.settingswindow Got value [["Bac d'alimentation", 'ADF']]
INFO   paperwork.frontend.settingswindow Resolutions found: [75, 100, 150, 200, 300, 600, 1200]
INFO   paperwork.frontend.settingswindow Selected source: 0
INFO   paperwork.frontend.settingswindow Selected source: 1
INFO   paperwork.frontend.settingswindow Got value [['75', 75]]
INFO   paperwork.frontend.settingswindow Got value [['100', 100]]
INFO   paperwork.frontend.settingswindow Got value [['150', 150]]
INFO   paperwork.frontend.settingswindow Got value [['200', 200]]
INFO   paperwork.frontend.settingswindow Got value [['300 (recommand\xc3\xa9e)', 300]]
INFO   paperwork.frontend.settingswindow Got value [['600', 600]]
INFO   paperwork.frontend.settingswindow Got value [['1200', 1200]]



I have just seen this difference:
INFO   paperwork.frontend.util.config Will scan using hpaio:/net/HP_LaserJet_Pro_MFP_M127fn?ip=192.168.1.14

INFO   paperwork.frontend.settingswindow Looking for resolution of device [hpaio:/net/HP_LaserJet_Pro_MFP_M127fn?ip=192.168.1.200]

Maybe a clue...

**Comment by 3like3beer** [comment]
OK if I change the default setting to "plateau" it works !!

**Comment by 3like3beer** [comment]
To recap : 

1. default =feeder / select scan from flatbed / nothing in the feeder : does not work (ko)
1. default =feeder / select scan from feeder / nothing in the feeder : does not work (ok)
1. default =flatbed / select scan from flatbed / nothing in the feeder : does work (ok)

1. default =* / select scan from feeder / something in the feeder : does work (ok)

**Comment by 3like3beer** [comment]
Ah ah. Ok, maybe you could rename the button scan as scan one page and scan
from feeder as scan multiple pages.
Btw, I would rather like to be able to sel ct the source with the button
and when I scan from feeder always get the multiple doc pop up, but this is
personal.

Le mar. 7 nov. 2017 à 08:09, Jerome Flesch <notifications@github.com> a
écrit :

**Comment by jflesch** [comment]
Yeah, changing this button is a discussion that has been going on for a long time : #537 , but it's pretty hard to find a solution that will satisfy at least 90% of users :(

Anyway, I'll use this ticket for the issue of having a popup saying "no scanner found" when it should say "nothing in the feeder" (or something like that).
I'll keep the UI changes for #537
